setup-disk: add 'sys' installation support for s390x

tmh1999 opened the pull request.

tmh1999 commented:
@jirutka Thanks

tmh1999 commented:
It seems like LVM support is not  so complicated. It will push update with LVM support in a couple of  hours.

tmh1999 commented:
https://github.com/alpinelinux/aports/pull/4061

jirutka commented on setup-disk.in:
```diff
@@ -459,6 +462,7 @@ install_mounted_root() {
 	case "$BOOTLOADER" in
 		grub) setup_grub "$mnt" "$root" "$modules" "$kernel_opts" "$bootdev" $disks ;;
 		syslinux) setup_syslinux "$mnt" "$root" "$modules" "$kernel_opts" "$bootdev" ;;
+		zipl) break;;
```
What loop are you trying to break here? o.O

tmh1999 replied:
Break from the case-esac :D Anyway that was updated in new commit.

jirutka commented on setup-disk.in:
```diff
@@ -994,6 +1006,21 @@ native_disk_install() {
+			# DASD devices do not support partition id
+			# assumptions were made that in 'sys' installation, partitions are in order: boot, swap, root
+			# TODO: mdev -s
+			cat > "$conf" <<- EOF
```
Please don’t put space between `<<-` and `EOF`.

tmh1999 replied:
Interesting. I have seen the differences between having space and not but can't remember it. Anyway it seems to work fine and I actually copied this from other place in the same file.

tmh1999 committed:
setup-disk: install on disk on s390x

In z/VM environment
ECKD DASDs need to be low-level formatted with dasdfmt and fdasd before
use. These devices don't have partition ids.
FBA DASDs are like normal disks - usable with sfdisk/fdisk - and have
partition ids.
Software raid and LVM on multiple devices are not supported at the
moment. Users could install with LVM on single disk and extend logical
volume to second disk.

In KVM environment
Virtual SCSI disks (virtio) are used which are like normal disks.

Bootloader is zipl from s390-tools package.

This commit introduces setup-dasd.in for DASD functions.

tmh1999 commented:
@ncopa we might need this in v3.8.

ncopa commented on setup-disk.in:
```diff
@@ -2,6 +2,7 @@
 
 PREFIX=
 . "$PREFIX/lib/libalpine.sh"
+. "$PREFIX/sbin/setup-dasd"
```
I don't think we should implement this as a `/sbin/setup-dasd`. Maybe better to have it as `/lib/dasd-functions.sh` or similar.

tmh1999 replied:
Agreed.

ncopa commented on setup-disk.in:
```diff
@@ -1266,6 +1307,7 @@ if is_efi || [ -n "$USE_EFI" ]; then
 fi
 
 [ "$ARCH" = "ppc64le" ] && BOOTLOADER=grub
+[ "$ARCH" = "s390x" ] && BOOTLOADER=zipl
```
I'd like:
````sh
case "$ARCH" in
ppc64le) BOOTLOADER=grub;;
s390x) BOOTLOADER=zipl;;
esac
````

tmh1999 replied:
Agreed.

tmh1999 commented:
Great! Let's merge then! Thank you.

tmh1999 commented:
Updated the name change to dasd-function.sh.in

tmh1999 commented:
@jirutka Hi, do you have any other comments on this one ?
